generators/go/internal/generator/sdk/core/stream.go

```diff
@@ -162,11 +162,15 @@ func (s *scannerStreamReader) parse(bytes []byte) (int, []byte, error) {
 		}
 	}
 	data := bytes[start:]
-	if i := strings.Index(string(data), s.options.delimiter); i >= 0 {
-		data = data[:i+len(s.options.delimiter)]
+	delimIndex := strings.Index(string(data), s.options.delimiter)
+	if delimIndex < 0 {
+		return len(bytes), data, nil
 	}
-	n := start + len(data) + len(s.options.delimiter)
-	return n, data, nil
+	endIdx := delimIndex + len(s.options.delimiter)
+	parsedData := data[:endIdx]
+	n := start + endIdx
+
+	return n, parsedData, nil
```

Comment:
nit: Thoughts on updating these variable names a bit so it's more consistent? Now that we're using the `*Index` suffix, we can do the same for `start` -> `startInde` (which would require a couple changes above):

```go
func (s *scannerStreamReader) parse(bytes []byte) (int, []byte, error) {
	var startIndex int
	if s.options != nil && s.options.prefix != "" {
		if i := strings.Index(string(bytes), s.options.prefix); i >= 0 {
			startIndex = i + len(s.options.prefix)
		}
	}
	data := bytes[startIndex:]
	delimIndex := strings.Index(string(data), s.options.delimiter)
	if delimIndex < 0 {
		return len(bytes), data, nil
	}
	endIndex := delimIndex + len(s.options.delimiter)
	parsedData := data[:endIndex]
	n := startIndex + endIndex
	return n, parsedData, nil
}
```

Reply:
Done! Good note.